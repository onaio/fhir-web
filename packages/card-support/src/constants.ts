// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
export const OPENSRP_URL_LOCATION_HIERARCHY = '/location/hierarchy';

/** CSV File MIME Types */
export const TEXT_CSV = 'text/csv';
export const APPLICATION_CSV = 'application/csv';
export const TEXT_PLAIN = 'text/plain';
export const TEXT_X_CSV = 'text/x-csv';
export const APPLICATION_VND_EXCEL = 'application/vnd.ms-excel';
export const APPLICATION_X_CSV = 'application/x-csv';
export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';
