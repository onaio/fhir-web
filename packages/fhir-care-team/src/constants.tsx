// url

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';

// FHIR API strings
export const URL_ADMIN = '/admin';
export const FHIR_CARE_TEAM = 'CareTeam';
export const FHIR_GROUPS = 'Group';
export const FHIR_PRACTITIONERS = 'Practitioner';
export const URL_EDIT_CARE_TEAM = `${URL_ADMIN}/CareTeam/edit`;
export const URL_CREATE_CARE_TEAM = `${URL_ADMIN}/CareTeam/new`;
export const URL_CARE_TEAM = `${URL_ADMIN}/CareTeams`;

// Route params
export const ROUTE_PARAM_CARE_TEAM_ID = 'careTeamId';
