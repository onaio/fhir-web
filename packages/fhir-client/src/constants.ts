export const dateFormat = 'yyyy-MM-dd';
