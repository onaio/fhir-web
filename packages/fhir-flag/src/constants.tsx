// urls
export const URL_ADMIN = '/admin';
export const URL_CLOSE_FLAGS = `${URL_ADMIN}/close-flags`;

// // magic strings

export const thatiMinutes = 30 * 60 * 1000;

// form fields
export const locationName = 'locationName' as const;
export const productName = 'productName' as const;
export const status = 'status' as const;
export const comments = 'comments' as const;
