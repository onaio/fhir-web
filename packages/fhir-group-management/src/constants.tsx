// url
export const ADD_EDIT_GROUP_URL = '/group/add-edit';
export const ADD_EDIT_COMMODITY_URL = '/commodity/add-edit';
export const LIST_GROUP_URL = '/groups/list';
export const LIST_COMMODITY_URL = '/commodity/list';

// magic strings
export const groupResourceType = 'Group';
export const listResourceType = 'List';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const type = 'type' as const;
export const unitOfMeasure = 'unitOfMeasure' as const;
