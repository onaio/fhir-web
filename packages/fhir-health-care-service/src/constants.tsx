// url
export const URL_ADD_HEALTHCARE = 'healthcare/add/';
export const URL_EDIT_HEALTHCARE = 'healthcare/edit/';

//
// Healthcares
//
export const HEALTH_CARE_SERVICE_RESOURCE_TYPE = 'HealthcareService';
export const HEALTHCARES_ENDPOINT = 'HealthcareService/';
