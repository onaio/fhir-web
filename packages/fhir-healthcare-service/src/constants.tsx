// url
export const ADD_EDIT_HEALTHCARE_SERVICE_URL = '/healthcare/add-edit';
export const LIST_HEALTHCARE_URL = '/healthcare/list';

//
// constants
//
export const healthCareServiceResourceType = 'HealthcareService';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const comment = 'comment' as const;
export const extraDetails = 'extraDetails' as const;
export const providedBy = 'providedBy' as const;
