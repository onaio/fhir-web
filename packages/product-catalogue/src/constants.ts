// opensrp api strings
export const OPENSRP_API_BASE_URL = 'https://mg-eusm-staging.smartregister.org/opensrp/rest/';
export const OPENSRP_PRODUCT_CATALOGUE = 'product-catalogue';
export const PRODUCT_ID_ROUTE_PARAM = 'productId';

// router routes
export const CATALOGUE_LIST_VIEW_URL = '/product-catalogue/list';
export const CATALOGUE_EDIT_VIEW_URL = '/product-catalogue/edit';
export const CATALOGUE_CREATE_VIEW_URL = '/product-catalogue/new';
export const HOME_URL = '/';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'product-catalogue';

/** route params for product-catalogue pages */
export interface RouteParams {
  productId?: string;
}
