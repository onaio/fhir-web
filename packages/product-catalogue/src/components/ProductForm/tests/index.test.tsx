/* eslint-disable @typescript-eslint/no-explicit-any */
import { mount, shallow } from 'enzyme';
import React from 'react';
import { ProductForm } from '..';
import { MemoryRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import toJson from 'enzyme-to-json';
import { product1 } from '../../../ducks/productCatalogue/tests/fixtures';
import { sendErrorNotification } from '@opensrp/notifications';
import { CATALOGUE_LIST_VIEW_URL } from '../../../constants';
import { product2, product3 } from './fixtures';
import * as opensrpReactUtils from '@opensrp/react-utils';

jest.mock('@opensrp/notifications', () => {
  return { sendSuccessNotification: jest.fn(), sendErrorNotification: jest.fn() };
});

jest.mock('@opensrp/react-utils', () => ({
  __esModule: true,
  ...Object.assign({}, jest.requireActual('@opensrp/react-utils')),
}));

// eslint-disable-next-line @typescript-eslint/no-var-requires
const fetch = require('jest-fetch-mock');

describe('productForm', () => {
  global.URL.revokeObjectURL = jest.fn();

  afterEach(() => {
    fetch.resetMocks();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it('renders without crashing', async () => {
    const wrapper = shallow(
      <MemoryRouter>
        <ProductForm />
      </MemoryRouter>
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });
  });

  it('renders correctly', async () => {
    fetch.once(JSON.stringify({ message: 'success' }));
    const div = document.createElement('div');
    document.body.appendChild(div);
    const wrapper = mount(
      <MemoryRouter>
        <ProductForm />
      </MemoryRouter>,
      { attachTo: div }
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    expect(wrapper.find('#productName label').text()).toMatchInlineSnapshot(`"Product name"`);
    expect(wrapper.find('#materialNumber label').text()).toMatchInlineSnapshot(`"Material number"`);
    wrapper.find('#isAttractiveItem label').forEach((radio, index) => {
      expect(radio.text()).toMatchSnapshot(`radio ${index}`);
    });

    expect(wrapper.find('#availability label').text()).toMatchInlineSnapshot(`"Is it there?"`);
    expect(wrapper.find('#condition label').text()).toMatchInlineSnapshot(
      `"Is it in good condition? (optional)"`
    );
    expect(wrapper.find('#appropriateUsage label').text()).toMatchInlineSnapshot(
      `"Is it being used appropriately? (optional)"`
    );
    expect(wrapper.find('#accountabilityPeriod label').text()).toMatchInlineSnapshot(
      `"Accountability period (in months)"`
    );
    expect(wrapper.find('#photoURL label').text()).toMatchInlineSnapshot(
      `"Photo of the product (optional)"`
    );
    expect(wrapper.find('#submit button').text()).toMatchInlineSnapshot(`"Submit"`);
  });

  it('validation works', async () => {
    fetch.once(JSON.stringify({ message: 'success' }));
    const div = document.createElement('div');
    document.body.appendChild(div);
    const wrapper = mount(
      <MemoryRouter>
        <ProductForm />
      </MemoryRouter>,
      { attachTo: div }
    );

    wrapper.find('form').simulate('submit');

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    expect((wrapper.find('FormItem#productName').props() as any).validateStatus).toEqual('error');
    expect(toJson((wrapper.find('FormItem#productName').props() as any).help)).toMatchSnapshot(
      `"Product name"`
    );

    expect((wrapper.find('FormItem#materialNumber').props() as any).validateStatus).toEqual(
      'error'
    );
    expect(toJson((wrapper.find('FormItem#materialNumber').props() as any).help)).toMatchSnapshot(
      `material number`
    );

    expect((wrapper.find('FormItem#isAttractiveItem').props() as any).validateStatus).toEqual(
      'error'
    );
    expect(toJson((wrapper.find('FormItem#isAttractiveItem').props() as any).help)).toMatchSnapshot(
      `isAttractive`
    );

    expect((wrapper.find('FormItem#availability').props() as any).validateStatus).toEqual('error');
    expect(toJson((wrapper.find('FormItem#availability').props() as any).help)).toMatchSnapshot(
      `availability`
    );

    expect((wrapper.find('FormItem#availability').props() as any).validateStatus).toEqual('error');
    expect(toJson((wrapper.find('FormItem#availability').props() as any).help)).toMatchSnapshot(
      `availability`
    );

    expect((wrapper.find('FormItem#condition').props() as any).validateStatus).toEqual(undefined);
    expect(toJson((wrapper.find('FormItem#condition').props() as any).help)).toMatchSnapshot(
      `condition`
    );

    expect((wrapper.find('FormItem#appropriateUsage').props() as any).validateStatus).toEqual(
      undefined
    );
    expect(toJson((wrapper.find('FormItem#appropriateUsage').props() as any).help)).toMatchSnapshot(
      `appropriateUsage`
    );

    expect((wrapper.find('FormItem#accountabilityPeriod').props() as any).validateStatus).toEqual(
      'error'
    );
    expect(
      toJson((wrapper.find('FormItem#accountabilityPeriod').props() as any).help)
    ).toMatchSnapshot(`accountability period`);

    expect((wrapper.find('FormItem#photoURL').props() as any).validateStatus).toEqual(undefined);
    expect(toJson((wrapper.find('FormItem#photoURL').props() as any).help)).toMatchSnapshot(
      `"photoURL"`
    );

    // fill form

    wrapper
      .find('input[name="productName"]')
      .simulate('change', { target: { name: 'productName', value: 'MotorCycle' } });

    wrapper
      .find('input[name="materialNumber"]')
      .simulate('change', { target: { name: 'materialNumber', value: 'MK-124' } });

    wrapper
      .find('input[type="radio"]')
      .first()
      .simulate('change', { target: { name: 'isAttractiveItem', checked: true } });

    wrapper
      .find('textarea[name="availability"]')
      .simulate('change', { target: { name: 'availability', value: 'Is available' } });

    wrapper
      .find('input[name="accountabilityPeriod"]')
      .simulate('change', { target: { name: 'accountabilityPeriod', value: '6' } });

    wrapper
      .find('textarea[name="condition"]')
      .simulate('change', { target: { name: 'condition', value: 'MotorCycle' } });

    wrapper
      .find('textarea[name="appropriateUsage"]')
      .simulate('change', { target: { name: 'appropriateUsage', value: 'MotorCycle' } });

    wrapper.find('form').simulate('submit');

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    // no errors, form submission completed, location changed to redirect url
    expect((wrapper.find('Router').props() as any).history.location.pathname).toEqual(
      CATALOGUE_LIST_VIEW_URL
    );
  });

  it('new form submission', async (done) => {
    fetch.once(JSON.stringify({ message: 'success' }));
    const div = document.createElement('div');
    document.body.appendChild(div);
    const wrapper = mount(
      <MemoryRouter>
        <ProductForm />
      </MemoryRouter>,
      { attachTo: div }
    );

    wrapper
      .find('input[name="productName"]')
      .simulate('change', { target: { name: 'productName', value: 'MotorCycle' } });

    wrapper
      .find('input[name="materialNumber"]')
      .simulate('change', { target: { name: 'materialNumber', value: 'MK-124' } });

    wrapper
      .find('input[type="radio"]')
      .first()
      .simulate('change', { target: { name: 'isAttractiveItem', checked: true } });

    wrapper
      .find('textarea[name="availability"]')
      .simulate('change', { target: { name: 'availability', value: 'Is available' } });

    wrapper
      .find('input[name="accountabilityPeriod"]')
      .simulate('change', { target: { name: 'accountabilityPeriod', value: '6' } });

    wrapper
      .find('textarea[name="condition"]')
      .simulate('change', { target: { name: 'condition', value: 'MotorCycle' } });

    wrapper
      .find('textarea[name="appropriateUsage"]')
      .simulate('change', { target: { name: 'appropriateUsage', value: 'MotorCycle' } });

    const file = new File([new ArrayBuffer(1)], 'file.jpg');
    wrapper.find('input[type="file"]').simulate('change', { target: { files: [file] } });
    wrapper.update();

    wrapper.find('form').simulate('submit');

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    const fd = fetch.mock.calls[0][1].body;
    const data = (Object as any).fromEntries(fd);

    // TODO - product photo does not update in setFieldValue call.

    const formFields = {
      productName: 'MotorCycle',
      materialNumber: 'MK-124',
      isAttractiveItem: true,
      condition: 'MotorCycle',
      appropriateUsage: 'MotorCycle',
      accountabilityPeriod: 6,
      availability: 'Is available',
    };
    expect(data).toMatchObject({
      file: expect.any(File),
      productCatalogue: expect.any(File),
    });

    const reader = new FileReader();
    reader.readAsText(data.productCatalogue);

    reader.addEventListener('load', function () {
      try {
        const result = reader.result;
        expect(JSON.parse(result as string)).toEqual(formFields);
        done();
      } catch (error) {
        done.fail(error);
      }
    });
  });

  it('edit form submission', async (done) => {
    fetch.once(JSON.stringify({ message: 'success' }));
    const div = document.createElement('div');
    document.body.appendChild(div);

    const props = {
      initialValues: product1,
    };

    const wrapper = mount(
      <MemoryRouter>
        <ProductForm {...props} />
      </MemoryRouter>,
      { attachTo: div }
    );

    // check that it has the initial correct values
    expect(toJson(wrapper.find('input[name="uniqueId"]'))).toMatchSnapshot('uniqueId');
    expect(wrapper.find('input[name="uniqueId"]').props().value).toEqual(1);

    expect(toJson(wrapper.find('input[name="productName"]'))).toMatchSnapshot('productName');
    expect(wrapper.find('input[name="productName"]').props().value).toEqual('Scale');

    expect(toJson(wrapper.find('input[name="materialNumber"]'))).toMatchSnapshot('materialNumber');
    expect(wrapper.find('input[name="materialNumber"]').props().value).toEqual('MT-123');

    expect(toJson(wrapper.find('input[type="radio"]').first())).toMatchSnapshot(
      'isAttractive radio1'
    );
    expect(toJson(wrapper.find('input[type="radio"]').last())).toMatchSnapshot(
      'isAttractive radio2'
    );

    expect(toJson(wrapper.find('textarea[name="availability"]'))).toMatchSnapshot('availability');
    expect(wrapper.find('textarea[name="availability"]').props().value).toEqual('available');

    expect(toJson(wrapper.find('input[name="accountabilityPeriod"]'))).toMatchSnapshot(
      'accountabilityPeriod'
    );
    expect(wrapper.find('input[name="accountabilityPeriod"]').props().value).toEqual('1');

    expect(toJson(wrapper.find('textarea[name="condition"]'))).toMatchSnapshot('condition');
    expect(wrapper.find('textarea[name="condition"]').props().value).toEqual('good condition');

    expect(toJson(wrapper.find('textarea[name="appropriateUsage"]'))).toMatchSnapshot(
      'appropriateUsage'
    );
    expect(wrapper.find('textarea[name="appropriateUsage"]').props().value).toEqual(
      'staff is trained to use it appropriately'
    );

    expect(wrapper.find('input[type="file"]')).toHaveLength(1);
    expect(wrapper.find('.upload-button').text()).toMatchSnapshot('Upload button');

    // change a few fields
    wrapper
      .find('input[name="productName"]')
      .simulate('change', { target: { name: 'productName', value: 'MotorCycle' } });

    wrapper
      .find('input[name="materialNumber"]')
      .simulate('change', { target: { name: 'materialNumber', value: 'MK-124' } });

    wrapper
      .find('input[type="radio"]')
      .first()
      .simulate('change', { target: { name: 'isAttractiveItem', checked: true } });

    wrapper
      .find('textarea[name="availability"]')
      .simulate('change', { target: { name: 'availability', value: 'Is available' } });

    wrapper
      .find('input[name="accountabilityPeriod"]')
      .simulate('change', { target: { name: 'accountabilityPeriod', value: '6' } });

    wrapper
      .find('textarea[name="condition"]')
      .simulate('change', { target: { name: 'condition', value: 'MotorCycle' } });

    wrapper
      .find('textarea[name="appropriateUsage"]')
      .simulate('change', { target: { name: 'appropriateUsage', value: 'MotorCycle' } });

    wrapper.find('form').simulate('submit');

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    const fd = fetch.mock.calls[0][1].body;

    const data = (Object as any).fromEntries(fd);

    const formFields = {
      productName: 'MotorCycle',
      materialNumber: 'MK-124',
      isAttractiveItem: true,
      condition: 'MotorCycle',
      appropriateUsage: 'MotorCycle',
      accountabilityPeriod: 6,
      availability: 'Is available',
      serverVersion: 123456,
      uniqueId: 1,
    };

    expect(data).toMatchObject({
      productCatalogue: expect.any(File),
    });

    const reader = new FileReader();
    reader.readAsText(data.productCatalogue);

    reader.addEventListener('load', function () {
      try {
        const result = reader.result;
        expect(JSON.parse(result as string)).toEqual(formFields);
        done();
      } catch (error) {
        done.fail(error);
      }
    });
  });

  it('Retest form submission formdata append error', async (done) => {
    // investigating formData.append: argument 2 is not an object error
    fetch.once(JSON.stringify({ message: 'success' }));
    const div = document.createElement('div');
    document.body.appendChild(div);

    const props = {
      initialValues: product2,
    };

    const wrapper = mount(
      <MemoryRouter>
        <ProductForm {...props} />
      </MemoryRouter>,
      { attachTo: div }
    );

    // change is it in good condition
    wrapper
      .find('textarea[name="condition"]')
      .simulate('change', { target: { name: 'condition', value: 'well maintained' } });

    wrapper.find('form').simulate('submit');

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    const fd = fetch.mock.calls[0][1].body;

    const data = (Object as any).fromEntries(fd);

    const formFields = {
      accountabilityPeriod: 2,
      appropriateUsage: 'this should be optional',
      availability: 'yeah',
      condition: 'well maintained',
      isAttractiveItem: false,
      materialNumber: 'asd',
      productName: 'Change name',
      uniqueId: 4,
    };

    expect(data).toMatchObject({
      productCatalogue: expect.any(File),
    });

    const reader = new FileReader();
    reader.readAsText(data.productCatalogue);

    reader.addEventListener('load', function () {
      try {
        const result = reader.result;
        expect(JSON.parse(result as string)).toEqual(formFields);
        done();
      } catch (error) {
        done.fail(error);
      }
    });
  });

  it('errors during edit form submission', async () => {
    const errorMessage = 'network error';
    fetch.mockReject(new Error(errorMessage));
    const div = document.createElement('div');
    document.body.appendChild(div);

    const errorNotificationSMock = jest.fn();
    (sendErrorNotification as jest.Mock).mockImplementation((...args) =>
      errorNotificationSMock(...args)
    );

    const props = {
      initialValues: product1,
    };

    const wrapper = mount(
      <MemoryRouter>
        <ProductForm {...props} />
      </MemoryRouter>,
      { attachTo: div }
    );

    wrapper.find('form').simulate('submit');

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    expect(errorNotificationSMock).toHaveBeenCalledWith('Error', errorMessage);
    (sendErrorNotification as jest.Mock).mockReset();
  });

  it('errors during new form submission', async () => {
    const otherErrorMessage = 'other network error';
    fetch.mockReject(new Error(otherErrorMessage));
    const div = document.createElement('div');
    document.body.appendChild(div);

    const errorNotificationSMock = jest.fn();
    (sendErrorNotification as jest.Mock).mockImplementation((...args) =>
      errorNotificationSMock(...args)
    );

    const wrapper = mount(
      <MemoryRouter>
        <ProductForm />
      </MemoryRouter>,
      { attachTo: div }
    );

    wrapper
      .find('input[name="productName"]')
      .simulate('change', { target: { name: 'productName', value: 'MotorCycle' } });

    wrapper
      .find('input[name="materialNumber"]')
      .simulate('change', { target: { name: 'materialNumber', value: 'MK-124' } });

    wrapper
      .find('input[type="radio"]')
      .first()
      .simulate('change', { target: { name: 'isAttractiveItem', checked: true } });

    wrapper
      .find('textarea[name="availability"]')
      .simulate('change', { target: { name: 'availability', value: 'Is available' } });

    wrapper
      .find('input[name="accountabilityPeriod"]')
      .simulate('change', { target: { name: 'accountabilityPeriod', value: '6' } });

    wrapper
      .find('textarea[name="condition"]')
      .simulate('change', { target: { name: 'condition', value: 'MotorCycle' } });

    wrapper
      .find('textarea[name="appropriateUsage"]')
      .simulate('change', { target: { name: 'appropriateUsage', value: 'MotorCycle' } });

    wrapper.find('form').simulate('submit');

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    expect(errorNotificationSMock).toHaveBeenCalledWith('Error', otherErrorMessage);
    (sendErrorNotification as jest.Mock).mockReset();
  });

  it('fetches image if photoURL is given', async () => {
    const fetchProtectedImageMock = jest
      .spyOn(opensrpReactUtils, 'fetchProtectedImage')
      .mockImplementation(() => Promise.resolve('hello'));

    const props = {
      initialValues: product3,
    };

    const wrapper = mount(
      <MemoryRouter>
        <ProductForm {...props} />
      </MemoryRouter>
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
    });
    wrapper.update();

    expect(fetchProtectedImageMock).toHaveBeenCalledWith(
      'https://mg-eusm-staging.smartregister.org/opensrp/multimedia/media/4'
    );
    expect(wrapper.find('img').get(0).props.src).toEqual('hello');
    wrapper.unmount();
    expect(global.URL.revokeObjectURL).toHaveBeenCalledWith('hello');
  });

  it('does not display image if fetch image fails', async () => {
    const fetchProtectedImageMock = jest
      .spyOn(opensrpReactUtils, 'fetchProtectedImage')
      .mockImplementation(() => Promise.resolve(null));

    const props = {
      initialValues: product3,
    };

    const wrapper = mount(
      <MemoryRouter>
        <ProductForm {...props} />
      </MemoryRouter>
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
    });
    wrapper.update();

    expect(fetchProtectedImageMock).toHaveBeenCalledWith(
      'https://mg-eusm-staging.smartregister.org/opensrp/multimedia/media/4'
    );
    expect(toJson(wrapper.find('img'))).toBeFalsy();

    wrapper.unmount();
  });

  it('display error toast if fetching image fails', async () => {
    const fetchProtectedImageMock = jest
      .spyOn(opensrpReactUtils, 'fetchProtectedImage')
      .mockImplementation(() => Promise.reject('error'));

    const props = {
      initialValues: product3,
    };

    const wrapper = mount(
      <MemoryRouter>
        <ProductForm {...props} />
      </MemoryRouter>
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
    });
    wrapper.update();

    expect(fetchProtectedImageMock).toHaveBeenCalledWith(
      'https://mg-eusm-staging.smartregister.org/opensrp/multimedia/media/4'
    );
    expect(toJson(wrapper.find('img'))).toBeFalsy();
    expect(sendErrorNotification).toHaveBeenCalledWith('Image could not be loaded');

    wrapper.unmount();
  });
});
