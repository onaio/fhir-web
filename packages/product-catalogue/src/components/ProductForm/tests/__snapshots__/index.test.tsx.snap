// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productForm edit form submission: Upload button 1`] = `"Upload"`;

exports[`productForm edit form submission: accountabilityPeriod 1`] = `
<input
  aria-valuemax={9007199254740991}
  aria-valuemin={0}
  aria-valuenow={1}
  autoComplete="off"
  className="ant-input-number-input"
  max={9007199254740991}
  min={0}
  name="accountabilityPeriod"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseUp={[Function]}
  required={false}
  role="spinbutton"
  step={1}
  value="1"
/>
`;

exports[`productForm edit form submission: appropriateUsage 1`] = `
<textarea
  className="ant-input"
  name="appropriateUsage"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Describe the product's intended use at the service point"
  rows={4}
  style={Object {}}
  value="staff is trained to use it appropriately"
/>
`;

exports[`productForm edit form submission: availability 1`] = `
<textarea
  className="ant-input"
  name="availability"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Describe where a supply monitor can locate this product at the service point."
  rows={4}
  style={Object {}}
  value="available"
/>
`;

exports[`productForm edit form submission: condition 1`] = `
<textarea
  className="ant-input"
  name="condition"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Describe how a supply monitor would assess whether the product is in good condition"
  rows={4}
  style={Object {}}
  value="good condition"
/>
`;

exports[`productForm edit form submission: isAttractive radio1 1`] = `
<input
  checked={true}
  className="ant-radio-input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  type="radio"
  value={true}
/>
`;

exports[`productForm edit form submission: isAttractive radio2 1`] = `
<input
  checked={false}
  className="ant-radio-input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  type="radio"
  value={false}
/>
`;

exports[`productForm edit form submission: materialNumber 1`] = `
<input
  className="ant-input"
  name="materialNumber"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  placeholder="Enter the product's material number"
  type="text"
  value="MT-123"
/>
`;

exports[`productForm edit form submission: productName 1`] = `
<input
  className="ant-input"
  name="productName"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  placeholder="Enter the product's name e.g Midwifery Kit"
  type="text"
  value="Scale"
/>
`;

exports[`productForm edit form submission: uniqueId 1`] = `
<input
  className="ant-input"
  id="uniqueId"
  name="uniqueId"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  readOnly={true}
  type="hidden"
  value={1}
/>
`;

exports[`productForm renders correctly: radio 0 1`] = `"Attractive item?"`;

exports[`productForm renders correctly: radio 1 1`] = `"yes"`;

exports[`productForm renders correctly: radio 2 1`] = `"no"`;

exports[`productForm validation works: "Product name" 1`] = `undefined`;

exports[`productForm validation works: "photoURL" 1`] = `undefined`;

exports[`productForm validation works: accountability period 1`] = `undefined`;

exports[`productForm validation works: appropriateUsage 1`] = `undefined`;

exports[`productForm validation works: availability 1`] = `undefined`;

exports[`productForm validation works: availability 2`] = `undefined`;

exports[`productForm validation works: condition 1`] = `undefined`;

exports[`productForm validation works: isAttractive 1`] = `undefined`;

exports[`productForm validation works: material number 1`] = `undefined`;
